Select correct webdriver version

MartinKavik opened the pull request:
Closes #611

**Changes**
- `webdriver.rs` split into `chromedriver.rs`, `geckodriver.rs` and `safaridriver.rs` for better code organization.
- Functions `install_geckodriver` and `install_chromedriver` fetches info about the latest `*driver` version so they can find out if we are using the latest `*driver`.

**Testing**
- I've tested it locally on Windows 10 - it downloads both drivers without problems.
- There are new / updated `curl` calls - should I test it in unit/integration tests somehow? (Is there an example for inspiration?)

**Possible future improvements**
- We can fetch info about a new `[chrome/gecko]driver` only when testing fails because of old driver and then restart testing.
  - **+** A little bit faster testing.
  - **+** We probably don't need internet connection during testing.
  - **+** Less requests to fetched endpoints.
  - **-** It's not possible to determine exact fail reason.
  - **-** We have to remember to restart it only once to prevent infinite loop.
  - **-** More complex code.
  - **-** Probably premature optimization because those fetches aren't noticeable during testing.
- Downloaded/cached drivers are never deleted/invalidated - we can delete old driver(s) when the newer one has been downloaded.
   - Probably out of scope this issue. Maybe general problem/feature of `wasm-pack` cache / crate `binary-install`.
- We can find out versions of the installed browsers and then download compatible drivers.
  - It's not super easy and we would need platform-specific code which can be error prone.
  - See comments for functions `get_chromedriver_url` and `get_geckodriver_url` for more info.

fitzgen commented:
Thanks for sending in this PR! Will take a look at this next week, after rust conf.

MartinKavik commented:
Changes in the latest commit (I can squash them before merging if you want):
- new file `src/stamps.rs` 
   - super simple key-value file store for saving versions + timestamps
- new file `tests/all/stamps.rs`
   - tests for `stamps.rs`
   - I've tried to write it as simple as possible but I'm waiting for review comments about best practices and file testing
   - added library `serial_test`
- updated `RELEASE_CHECKLIST` so we don't forget to update driver versions
- updated `webdrivers*`
  - see comment above function `get_chromedriver_url` or  `get_geckodriver_url` - there is a new algorithm for getting driver version (it should be similar to your comment - check once per day, "offline" mode, etc.)
- added `context`s - I'm waiting for review comments about wording and `failure` / error handling best practices

I've tested it with a local project and it seems working even without internet connection (if drivers are already downloaded).

Your comments from the previous review should be resolved.
Thanks for reviews!
